pkg/flink/internal/store/store.go

```diff
@@ -105,6 +105,9 @@ func (s *Store) ProcessStatement(statement string) (*types.ProcessedStatement, *
 }
 
 func createSqlV1Statement(statement string, statementName string, computePoolId string, properties map[string]string) flinkgatewayv1.SqlV1Statement {
+	if !maybeCreateAlterModelStatement(statement) {
```

Comment:
Would it make sense to only send those secret properties that are actually referenced by the SQL, or does ALTER MODEL need them even if they aren't listed?

Reply:
alter model can also reference it. Make sense to check they are used